libgraphics/include/pivot/graphics/vk_utils.hxx

```diff
@@ -7,58 +7,48 @@
 #include <vulkan/vulkan.hpp>
 
 #include "pivot/graphics/VulkanException.hxx"
+#include "pivot/graphics/types/AllocatedBuffer.hxx"
 
-namespace pivot::graphics::vk_utils
+namespace pivot::graphics
 {
-template <typename T>
-concept is_copyable = requires
-{
-    std::is_standard_layout_v<T>;
-    typename std::vector<T>;
-};
 
-constexpr void vk_try(vk::Result err)
+namespace vk_utils
 {
-    if (err < vk::Result::eSuccess) throw VulkanException(err);
-}
-
-constexpr void vk_try(VkResult res) { vk_try(vk::Result(res)); }
+    template <typename T>
+    concept is_copyable = requires
+    {
+        std::is_standard_layout_v<T>;
+        typename std::vector<T>;
+    };
 
-template <class... FailedValue>
-constexpr bool vk_try_mutiple(const vk::Result result, const FailedValue... failedResult)
-{
-    if (((result == failedResult) || ...)) {
-        return true;
-    } else {
-        vk_try(result);
-        return false;
+    constexpr void vk_try(vk::Result err)
+    {
+        if (err < vk::Result::eSuccess) throw VulkanException(err);
     }
-}
 
-std::vector<std::byte> readFile(const std::string &filename);
-vk::ShaderModule createShaderModule(const vk::Device &device, const std::vector<std::byte> &code);
+    constexpr void vk_try(VkResult res) { vk_try(vk::Result(res)); }
 
-vk::SampleCountFlagBits getMaxUsableSampleCount(vk::PhysicalDevice &physical_device);
-vk::Format findSupportedFormat(vk::PhysicalDevice &gpu, const std::vector<vk::Format> &candidates,
-                               vk::ImageTiling tiling, vk::FormatFeatureFlags features);
-bool hasStencilComponent(vk::Format format) noexcept;
-
-namespace tools
-{
-    template <typename T>
-    void print_array(Logger::Stream stream, const std::string_view &message, const std::vector<T> &val)
+    template <class... FailedValue>
+    constexpr bool vk_try_mutiple(const vk::Result result, const FailedValue... failedResult)
```

Comment:
Can you document what this function does ? I have a hard time understanding what it is doing.